src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -15,34 +16,87 @@ import {ICoExpressionPlotProps} from "./CoExpressionPlot";
 import {getMobxPromiseGroupStatus} from "../../../shared/lib/getMobxPromiseGroupStatus";
-import MolecularProfileSelector from "../../../shared/components/MolecularProfileSelector";
+import {IDataQueryFilter} from "shared/lib/StoreUtils";
 
 export interface ICoExpressionTabProps {
     store:ResultsViewPageStore;
 }
 
-export class CoExpressionCache extends MobxPromiseCache<{entrezGeneId:number, molecularProfile:MolecularProfile, allData:boolean}, CoExpression[]> {};
+export class CoExpressionCache extends MobxPromiseCache<{profileX: MolecularProfile, 
+    profileY: MolecularProfile, geneticEntityId: string,
+    geneticEntityType: string, allData:boolean}, CoExpression[]> {}
 
 @observer
 export default class CoExpressionTab extends React.Component<ICoExpressionTabProps, {}> {
-    @observable _selectedMolecularProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
-    @observable _selectedEntrezGeneId:number | undefined; // only undefined initially, until genes downloaded
+    @observable _selectedProfileXGene:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedProfileXGeneset:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedProfileY:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedGeneticEntityId:string | undefined; // only undefined initially, until genes and gene sets downloaded
+    @observable _selectedGeneticEntityName:string| undefined; // only undefined initially, until genes and gene sets downloaded
+    @observable _isSelectedGeneticEntityAGeneSet: boolean;
 
-    @computed get selectedMolecularProfile():MolecularProfile|undefined {
-        if (!this._selectedMolecularProfile && this.props.store.coexpressionTabMolecularProfiles.isComplete
-            && this.props.store.coexpressionTabMolecularProfiles.result.length > 0) {
-            return this.props.store.coexpressionTabMolecularProfiles.result[0];
+    @computed get selectedProfileXGene():MolecularProfile|undefined {
+        if (!this._selectedProfileXGene && this.props.store.coexpressionTabMolecularProfilesXGene.isComplete
+                && this.props.store.coexpressionTabMolecularProfilesXGene.result.length > 0) {
+                    return this.props.store.coexpressionTabMolecularProfilesXGene.result[0];
         } else {
-            return this._selectedMolecularProfile;
+            return this._selectedProfileXGene;
         }
     }
 
-    @computed get selectedEntrezGeneId():number | undefined {
-        if (this._selectedEntrezGeneId === undefined && this.props.store.genes.isComplete &&
-                this.props.store.genes.result.length > 0) {
-            return this.props.store.genes.result[0].entrezGeneId;
+    @computed get selectedProfileXGeneset():MolecularProfile|undefined {
```

Comment:
like Ive said in other places, Im really not convinced we need to handle these cases separately - it's always a MolecularProfile, and always exactly one of them is used. logically it occupies one space, not two